test/cpp/end2end/xds/xds_security_end2end_test.cc

```diff
@@ -153,15 +153,16 @@ class FakeCertificateProvider final : public grpc_tls_certificate_provider {
             "No certificates available for cert_name \"", cert_name, "\""));
         distributor_->SetErrorForCert(cert_name, error, error);
       } else {
-        std::optional<std::string> root_certificate;
+        std::shared_ptr<RootCertInfo> root_cert_info;
         std::optional<grpc_core::PemKeyCertPairList> pem_key_cert_pairs;
         if (root_being_watched) {
-          root_certificate = it->second.root_certificate;
+          root_cert_info =
+              std::make_shared<RootCertInfo>(it->second.root_certificate);
```

Comment:
Please add a test with a SPIFFE bundle map.

Reply:
This is coming in the next PR - we can't end2end test the spiffe bundle map until the actual verification logic using it is in